### PR TITLE
add a setup k3d action

### PR DESCRIPTION
**Pull request by joshrwolf**
alternative to `setup-kind`, this time with more `k3s` and faster startups

also adds a more generic `k8s-diag` to replace the purpose built `kind-diag`

### CODE REVIEW

File: setup-k3d/action.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Copyright 2022 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Setup k3d'
+description: |
+  This action sets up a k3d cluster based on the supplied
+  configuration knobs.
+
+inputs:
+  k3s-version:
```
Comment (joshrwolf) [inline setup-k3d/action.yaml:10]: make this `k3s-image` instead

File: setup-k3d/action.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    description: |
+      The version of k3s to use in the form: 1.27.x
+    required: false
+    default: 1.27.x
```
Comment (mattmoor) [inline setup-k3d/action.yaml:14]: I'd replace this with the image tag to use and default to `cgr.dev/chainguard/k3s:latest`

File: setup-k3d/action.yaml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+          #   echo "K3S_IMAGE" >> $GITHUB_ENV
+          #   ;;
+
+          # TODO: We only have a cgr image for the latest k3s version for now
+          1.27.x)
+            echo "K3S_IMAGE=cgr.dev/chainguard/k3s@sha256:ab900ffe02fc1263b671c801b3cfc6070db76eaf5a57ee0d31e186d821d49c44" >> $GITHUB_ENV
+            ;;
+
+          *) echo "Unsupported version: ${{ inputs.k3s-version }}"; exit 1 ;;
+        esac
```
Comment (mattmoor) [inline setup-k3d/action.yaml:104]: I'd just kill this.

File: k8s-diag/action.yaml [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# Copyright 2022 Chainguard, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Collect KinD diagnostics'
```
Comment (mattmoor) [inline k8s-diag/action.yaml:4]: lol scrub KinD

File: setup-k3d/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Setup KinD
```
Comment (mattmoor) [inline setup-k3d/README.md:1]: k3d :)

**Review by mattmoor — APPROVED** [review]

File: setup-k3d/README.md (hunk 5 above)
Comment (mattmoor) [inline setup-k3d/README.md:1]:
```suggestion
# Setup K3d
```

**Pull request merged**